python/tvm/contrib/hexagon/session.py

```diff
@@ -88,14 +88,23 @@ def __enter__(self):
                     self._rpc_receive_buffer_size_bytes,
                 ],
             )
+            func = self._rpc.get_function("device_api.hexagon.acquire_resources")
+            func()
             return self
 
         except RuntimeError as exception:
             raise exception
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        # close session to the tracker
-        del self._rpc
+        try:
+            func = self._rpc.get_function("device_api.hexagon.release_resources")
+            func()
+        except RuntimeError:
+            # do nothing, as we are shutting down
```

Comment:
we could probably `_LOG.warn("Exception occurred while calling release_resources() during Session __exit__", exc_info=1)` here

Reply:
We don't have access to _LOG here, but I added a print.